integration-libs/opf/payment/core/services/opf-payment-hosted-fields.service.ts

```diff
@@ -237,4 +200,22 @@ export class OpfPaymentHostedFieldsService {
       );
     }
   }
+  protected getCartAccessCode(
+    submitRequest: SubmitRequest | SubmitCompleteRequest
+  ): Observable<
+    [SubmitRequest | SubmitCompleteRequest, { accessCode: string }]
+  > {
+    return combineLatest([
+      this.userIdService.takeUserId(),
+      this.activeCartFacade.takeActiveCartId(),
+    ]).pipe(
+      switchMap(([userId, activeCartId]: [string, string]) => {
+        submitRequest.cartId = activeCartId;
+        return combineLatest([
+          of(submitRequest),
+          this.cartAccessCodeFacade.getCartAccessCode(userId, activeCartId),
+        ]);
+      })
+    );
+  }
```

Comment:
Note that I removed the take(1) as takeUserId and takeActiveCartId already have 1. and getCartAccessCode is an ajax call which alwats get completed